tests/appsec/test_automated_login_events.py

```diff
@@ -1348,3 +1218,703 @@ def validate_iden(meta):
+    "datadog/2/ASM_DD/rules/config",
+    {
+        "version": "2.1",
+        "metadata": {"rules_version": "1.2.6"},
+        "rules": [
+            {
+                "id": "block-users",
+                "name": "Block User Addresses",
+                "tags": {"type": "block_user", "category": "security_response"},
+                "conditions": [
+                    {
+                        "parameters": {"inputs": [{"address": "usr.login"}], "data": "blocked_users"},
+                        "operator": "exact_match",
+                    }
+                ],
+                "transformers": [],
+                "on_match": ["block"],
+            }
+        ],
+        "rules_data": [
+            {
+                "id": "blocked_users",
+                "type": "data_with_expiration",
+                "data": [{"value": "social-security-id", "expiration": 0}, {"value": "sdkUser", "expiration": 0}],
+            },
+        ],
+    },
+)
+
+
+@rfc("https://docs.google.com/document/d/1RT38U6dTTcB-8muiYV4-aVDCsT_XrliyakjtAPyjUpw")
+@features.user_monitoring
+@scenarios.appsec_runtime_activation
+class Test_V3_Login_Events_Blocking:
+    def setup_login_event_blocking_auto(self):
+        rc.rc_state.reset().apply()
+
+        self.config_state_1 = rc.rc_state.set_config(*CONFIG_ENABLED).apply()
+        self.r_login = weblog.post("/login?auth=local", data=login_data(context, USER, PASSWORD))
+
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_USER).apply()
+        self.r_login_blocked = weblog.post("/login?auth=local", data=login_data(context, USER, PASSWORD))
+
+    def test_login_event_blocking_auto(self):
+        assert self.config_state_1[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.r_login.status_code == 200
+
+        assert self.config_state_2[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        interfaces.library.assert_waf_attack(self.r_login_blocked, rule="block-users")
+        assert self.r_login_blocked.status_code == 403
+
+    def setup_login_event_blocking_sdk(self):
+        rc.rc_state.reset().apply()
+
+        self.config_state_1 = rc.rc_state.set_config(*CONFIG_ENABLED).apply()
+        self.r_login = weblog.post(
+            "/login?auth=local&sdk_event=success&sdk_user=sdkUser", data=login_data(context, UUID_USER, PASSWORD)
+        )
+
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_USER).apply()
+        self.r_login_blocked = weblog.post(
+            "/login?auth=local&sdk_event=success&sdk_user=sdkUser", data=login_data(context, UUID_USER, PASSWORD)
+        )
+
+    def test_login_event_blocking_sdk(self):
+        assert self.config_state_1[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.r_login.status_code == 200
+
+        assert self.config_state_2[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        interfaces.library.assert_waf_attack(self.r_login_blocked, rule="block-users")
+        assert self.r_login_blocked.status_code == 403
```

Comment:
this test is a bit unclear, are you trying to check if it's blocking on usr.login ? are you trying to check if it's blocking on the SDK ? are you trying to check if it's blocking correctly in user id after the SDK ?

If you're trying to check for login, then you need to change the RC payload because login is `testuuid`, not `social-security-id`.
If you're trying to check for the SDK, it's not the right place to test this IMO, this is the automated login events tests.
If you're trying to check that it's automatically blocking correctly even with the SDK, then it might depend on if the SDK was called before or after the framework.